app/assets/javascripts/views/footer/footer_view.ts

```diff
@@ -540,9 +540,11 @@ class FooterViewCtrl extends PureViewCtrl<
     const id = 'dock-svg-' + shortcut.component.uuid;
     const element = document.getElementById(id)!;
     const parser = new DOMParser();
-    const svg = shortcut.component.package_info.dock_icon.source;
-    const doc = parser.parseFromString(svg, 'image/svg+xml');
-    element.appendChild(doc.documentElement);
+    const svg = shortcut.component.package_info.dock_icon?.source;
+    if(svg!=null) {
```

Comment:
Be sure to set up auto-lint on save, or lint before committing.
```suggestion
    if (svg != null) {
```

Reply:
Should we check for undefined here too?